modcc/expression.cpp

```diff
@@ -303,12 +308,20 @@ expression_ptr ReactionExpression::clone() const {
 }
 
 void ReactionExpression::semantic(scope_ptr scp) {
+    error_ = false;
     scope_ = scp;
+
     lhs()->semantic(scp);
     rhs()->semantic(scp);
 
     fwd_rate()->semantic(scp);
     rev_rate()->semantic(scp);
+
+    if(lhs_->has_error() || rhs_->has_error() || fwd_rate_->has_error() || rev_rate_->has_error()) {
```

Comment:
Do we need to set the error state here? The error visitor will collect the errors set in `lhs`, `rhs`, etc. An additional error message here won't make much sense?

Reply:
We can just set the error flag and keep the same error message.

Reply:
The error visitor will only collect the errors of the main expression, in this case the `ReactionExpression`. The errors of `lhs`, `rhs`, `fwd_rate` and `rev_rate` will not be collected by the error visitor unless they are propagated to the main `ReactionExpression` expression.

The `ErrorVisitor` has visitors for each of: 
* `Expression`
* `ProcedureExpression`
* `FunctionExpression`
* `UnaryExpression`
* `BinaryExpression`
* `CallExpression`
* `BlockExpression`
* `InitialBlock`
* `IfExpression`

If we don't set the error state of the `ReactionExpression`: when the `ErrorVisitor` gets to it, it will call the default `Expression` visitor which won't print any errors, since the `ReactionExpression` itself has no errors.

If we don't want to propagate the errors of subexpressions (`lhs`, `rhs` etc) to their calling expressions, then we can extend the error visitor to visit a wider set of expressions and recursively check all subexpressions for errors.